### PR TITLE
PR adds SKILLOUT_PROGRESSIVE and changes food to 1/3/5 1/2/3

### PR DESCRIPTION
**Pull request by artyomtrityak**

### CODE REVIEW

File: items.h [hunk 1]
```diff
@@ -96,7 +96,8 @@ class ItemType
 			// This item cannot be transported.
 			NOTRANSPORT = 0x20,
 			// Produced monsters
-			MANPRODUCE = 0x40
+			MANPRODUCE = 0x40,
+			SKILLOUT_PROGRESSIVE = 0x50
```
Comment (valdisz) [inline items.h:100]: We have similar feature in other skills and it is named differently. From the name "progressive" is not clear what it does.

**Review by valdisz — CHANGES_REQUESTED** [review]

File: monthorders.cpp [hunk 2]
```diff
@@ -1178,6 +1180,15 @@ void Game::RunUnitProduce(ARegion * r,Unit * u)
...
 		output *= level;
+	if (ItemDefs[o->item].flags & ItemType::SKILLOUT_PROGRESSIVE) {
+		if (level == 3) {
+			output *= 2;
+		}
+		if (level == 5) {
+			output *= 3;
+		}
+	}
+
```
Comment (valdisz) [inline monthorders.cpp:1191]: What you wanted to achieve here? To have skill level rounded up produced amount just do: (level + 1) / 2

Then results will be following:
L1 = 1 item
L2 = 1 item
L3 = 2 items
L4 = 2 items
L5 = 3 items

`output *= (level + 1) / 2`

File: skillshows.cpp [hunk 3]
```diff
@@ -1631,6 +1631,10 @@ AString *ShowSkill::Report(Faction *f)
 				if (ItemDefs[i].flags & ItemType::SKILLOUT) {
 					temp1 += " equal to their skill level";
 				}
+				if (ItemDefs[i].flags & ItemType::SKILLOUT_PROGRESSIVE) {
+					temp1 += " based on their skill level. Level 1 produces 1 item, ";
+					temp1 += " level 3 produces 2 items, level 5 produces 3 items";
```
Comment (valdisz) [inline skillshows.cpp:1636]: Then here is easy to make generic examples w/o hardcoding.

**Review by valdisz — CHANGES_REQUESTED** [review]

File: items.h (hunk 1 above)
Comment (valdisz) [reply items.h:100]: This is an output multiplier. Maybe use flag name that reflects such behaviour?

Comment (artyomtrityak) [reply items.h:100]: sure, I will change name

File: monthorders.cpp (hunk 2 above)
Comment (artyomtrityak) [reply monthorders.cpp:1191]: sounds good

**Review by valdisz — APPROVED** [review]

**Pull request merged**